packages/react-hooks/src/hooks/toggle.ts

```diff
@@ -1,16 +1,31 @@
 import {useState, useCallback} from 'react';
 
+type ToggleReturn = {
+  value: boolean;
+  setTrue: () => void;
+  setFalse: () => void;
+  toggle: () => void;
+} & [boolean, () => void, () => void, () => void];
```

Comment:
Ah interesting! If I understand this correctly, although you're using an Intersection type here, the `ToggleReturn` type can only be either one of the two types you've defined here since there's no way to overlap the two. But this only really works because, like you mentioned, an array is a special type of object.. is this common practice?